File: rust/src/leetcode/Q134_gas_station.rs
/*
 * @lc app=leetcode id=134 lang=rust
 *
 * [134] Gas Station
 */

// @lc code=start
impl Solution {
    pub fn can_complete_circuit(gas: Vec<i32>, cost: Vec<i32>) -> i32 {
        if gas.iter().sum::<i32>() < cost.iter().sum::<i32>() {
            return -1;
        }

        let mut start: usize = 0;
        let mut current_gas: i32 = 0;

        for i in 0..gas.len() {
            current_gas += gas[i] - cost[i];

            if current_gas < 0 {
                start = i + 1;
                current_gas = 0;
            }
        }

        start as i32
    }
}
// @lc code=end

File: python/Q134_gas_station.py
#
# @lc app=leetcode id=134 lang=python3
#
# [134] Gas Station
#


# @lc code=start
class Solution:
    def canCompleteCircuit(self, gas: list[int], cost: list[int]) -> int:
        if sum(gas) < sum(cost):
            return -1

        start: int = 0
        current_gas: int = 0

        for i in range(len(gas)):
            current_gas += gas[i] - cost[i]

            if current_gas < 0:
                start = i + 1
                current_gas = 0

        return start


# @lc code=end
